### PR TITLE
Add Support for Nested Function in Order By Clause

### PR DESCRIPTION
**Pull request by forestmvey**
### Description
Add support for nested function use in the `ORDER BY` clause.
 
### Example Queries
```sql
SELECT nested(message.info) from nested_objects ORDER BY nested(message.info);
```

### Issues Resolved
Portion of Issue: [1111](https://github.com/opensearch-project/sql/issues/1111)
Note this does not resolve issue [1777](https://github.com/opensearch-project/sql/issues/1788) but only bring V2 engine to parity with V1 engine.
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/sort/SortQueryBuilder.java [hunk 1]
```diff
@@ -53,11 +59,44 @@ public SortBuilder<?> build(Expression expression, Sort.SortOption option) {
...
       throw new IllegalStateException("unsupported expression " + expression.getClass());
     }
   }
 
+  /**
+   * Validate semantics for arguments in nested function.
+   * @param nestedFunc Nested function expression.
+   */
+  private void validateNestedArgs(FunctionExpression nestedFunc) {
+    if (nestedFunc.getArguments().size() > 2) {
```
Comment (acarbonetto) [inline opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/sort/SortQueryBuilder.java:83]: do we also need to check for `size() < 1`?

File: opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/sort/SortQueryBuilderTest.java [hunk 2]
```diff
@@ -25,6 +28,62 @@ void build_sortbuilder_from_reference() {
...
+                DSL.ref("message", STRING),
+                DSL.ref("message", STRING)
+            ),
+            Sort.SortOption.DEFAULT_ASC
+        )
+    );
+  }
+
+  @Test
+  void nested_with_invalid_arg_type_throws_exception() {
```
Comment (acarbonetto) [inline opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/sort/SortQueryBuilderTest.java:67]: do we want to add a test case to validate that `nested()` returns an IllegalArgumentException?

**Review by acarbonetto — DISMISSED** [review]
Please add another check for empty nested call if that's possible.  
Otherwise, we should add a parser test/check that fails to parse

**Review by matthewryanwells — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by GumpacG — APPROVED** [review]

File: opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/sort/SortQueryBuilderTest.java (hunk 2 above)
Comment (forestmvey) [reply opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/sort/SortQueryBuilderTest.java:67]: Already have one [here](https://github.com/opensearch-project/sql/blob/84fab78dce84f89befa6012acc55792c2078892f/opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/sort/SortQueryBuilderTest.java#L52-L64)

File: opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/sort/SortQueryBuilder.java (hunk 1 above)
Comment (forestmvey) [reply opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/sort/SortQueryBuilder.java:83]: Yes I'll add this check and a relevant test too.

**Review by MitchellGale — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by GumpacG — APPROVED** [review]

**Pull request merged**